blockauth/serializers/wallet_serializers.py

```
@@ -48,16 +42,16 @@ def validate(self, data):
             authenticator = WalletAuthenticator()
             if not authenticator.verify_signature(wallet_address, message, signature):
                 raise ValidationError(
-                    detail={"signature": "Invalid signature. Signature verification failed."}, code=4009
+                    detail={"signature": "Invalid signature. Signature verification failed."}, code="INVALID_SIGNATURE"
                 )
         except ValueError as e:
             # Structured validation errors from replay/timestamp/nonce checks
-            raise ValidationError(detail={"message": str(e)}, code=4009)
+            raise ValidationError(detail={"message": str(e)}, code="INVALID_SIGNATURE")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)